provider/local/provider.go

```diff
@@ -36,6 +36,9 @@ type Provider interface {
 
 	// Returns a configured ContextCredentialsFactory for this provider
 	ContextCredentialsFactory(datacenter *string) (ContextCredentialsFactory, error)
+
+	// UpdateAPIKey updates api key in the provider
+	UpdateAPIKey(*vpcconfig.VPCBlockConfig, *zap.Logger) error
```

Comment:
In case of file are planning to introduce one more variant of UpdateAPIKey as it will be using different config.

Reply:
I have an idea for the same. Will raise another PR for this

Reply:
Ok we can take this as improvment